cmds/dhclient/dhcp6client/dhcp6client.go

```diff
@@ -0,0 +1,138 @@
+package dhcp6client
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"time"
+
+	"github.com/mdlayher/dhcp6"
+	"github.com/vishvananda/netlink"
+)
+
+type Client struct {
+	hardwareAddr  net.HardwareAddr //The HardwareAddr to send in the request.
+	ignoreServers []net.IP         //List of Servers to Ignore requests from.
+	timeout       time.Duration    //Time before we timeout.
+	broadcast     bool             //Set the Bcast flag in BOOTP Flags
+	connection    connection       //The Connection Method to use
+}
+
+/*
+*  * Abstracts the type of underlying socket used
+*   */
+type connection interface {
+	Close() error
+	Write(packet []byte) error
+	ReadFrom()
+	// SetReadTimeout(t time.Duration) error
+}
+
+func New(haddr net.HardwareAddr, conn connection, timeout time.Duration) (*Client, error) {
```

Comment:
@hugelgupf Thanks a lot for the comments.

What I am trying to do now is to firstly wrap up a packet. After that, by mimicking what's been done for hdcpv4, I send out then packet and see if I am able to get a reasonable response at least. So everything is a draft at this moment, though I do realize that I need more strict checks on errors, etc.

What is happening is, I send out a packet but get a response with a nonsense message type. I figured this out by printing out the raw packet bytes. No errors have been thrown in between.